packages/vscode-extension/src/treeview/environmentTreeItem.ts

```diff
@@ -0,0 +1,233 @@
+    if (envName === LocalEnvironmentName) {
+      return (await isExistingTabApp(ext.workspaceUri.fsPath))
+        ? EnvInfo.LocalForExistingApp
+        : EnvInfo.Local;
+    } else {
+      const provisionSucceeded = await getProvisionSucceedFromEnv(envName);
+      return provisionSucceeded ? EnvInfo.ProvisionedRemoteEnv : EnvInfo.RemoteEnv;
+    }
+  }
+}
+
+class WarningNode extends DynamicNode {
+  constructor(public identifier: string, warnings: string[]) {
+    super(identifier, vscode.TreeItemCollapsibleState.None);
+    this.label = `Sign in with your correct Azure / M365 account`;
+    this.iconPath = warningIcon;
+    this.tooltip = this.formatWarningMessages(warnings);
+  }
+
+  public async getChildren(): Promise<DynamicNode[] | undefined | null> {
+    return null;
+  }
+
+  public async getTreeItem(): Promise<vscode.TreeItem> {
+    return this;
+  }
+
+  private formatWarningMessages(warnings: string[]): string {
+    let warningMessage = "";
+    if (warnings.length > 1) {
+      const formattedWarnings = warnings.map((warning) => "> ".concat(warning));
+      warningMessage = formattedWarnings.join("\n");
+    } else {
+      warningMessage = warnings[0];
+    }
+
+    return warningMessage;
+  }
+}
+
+class SubscriptionNode extends DynamicNode {
+  private resourceGroupNode?: ResourceGroupNode;
+  constructor(public identifier: string, private subscriptionInfo: SubscriptionInfo) {
+    super(identifier, vscode.TreeItemCollapsibleState.None);
+    this.contextValue = "openSubscriptionInPortal";
+    this.iconPath = subscriptionIcon;
+  }
+
+  public async getChildren(): Promise<DynamicNode[] | undefined | null> {
+    if (this.resourceGroupNode) {
+      return [this.resourceGroupNode];
+    }
+    return null;
+  }
+
+  public async getTreeItem(): Promise<vscode.TreeItem> {
+    this.tooltip = this.subscriptionInfo.subscriptionName
+      ? util.format(
+          localize("teamstoolkit.envTree.subscriptionTooltip"),
+          this.identifier,
+          this.subscriptionInfo.subscriptionName,
+          this.subscriptionInfo.subscriptionId
+        )
+      : util.format(
+          localize("teamstoolkit.envTree.subscriptionTooltipWithoutName"),
+          this.identifier,
+          this.subscriptionInfo.subscriptionId
+        );
+    this.label = this.subscriptionInfo.subscriptionName ?? this.subscriptionInfo.subscriptionId;
+    this.description = this.subscriptionInfo.subscriptionId;
+
+    const resourceGroupName = await getResourceGroupNameFromEnv(this.identifier);
+    if (resourceGroupName) {
+      this.resourceGroupNode = new ResourceGroupNode(this.identifier, resourceGroupName);
+      this.collapsibleState = vscode.TreeItemCollapsibleState.Expanded;
+    } else {
+      this.resourceGroupNode = undefined;
+      this.collapsibleState = vscode.TreeItemCollapsibleState.None;
+    }
+
+    return this;
+  }
+}
+
+class ResourceGroupNode extends DynamicNode {
```

Comment:
where's the collaboration related node?

Reply:
Do you mean `generateCollaboratorNode()` and `generateCollaboratorWarningNode()`? I didn't find any usage so I just removed these 2 functions.
Other collaboration functions seem to be provided by "view item context" and they are updated in `extension.ts`.

Reply:
@SLdragon @KennethBWSong may take a look at the usage of collaboration node.

Reply:
Thanks, we don't add collaboration node to the tree view any more, so we can safely remove this two function